### DISCUSSION TITLE
how to make changes to my book

### DISCUSSION THREAD

**Discussion by chuymtz** [executablebooks/meta#786] [General]
Im very new to this package. I've tried looking for a similar question but can't find it. 
What's the reason behind having a local and online version? Is the idea that one should not be making changes to the book once it's published?

**Comment by rowanc1** [comment]
You use jupyter book by working with markdown and jupyter notebook files on your local computer, often using github as a collaboration platform. You then deploy the changes to an online website. The local and online versions are often the same, and it is just the process of developing the book that starts out on your computer and then is deployed later.

**Reply by chuymtz** [reply]
Thanks for the reply. I think my confusion comes from reading the documentation carelessly. It suggests that gh-import creeated the new branch but I can't see it. So I made one and now I'm confused as to what a normal workflow should be. Shouldi edit my book in main and merge to gh-pages?

**Reply by rowanc1** [reply]
You shouldn't ever have to edit content on gh-pages, instead you should use an action like this:
https://jupyterbook.org/en/stable/publish/gh-pages.html#automatically-host-your-book-with-github-actions
When you commit to `main`, the action executes/builds your book, and then copies the result onto the `gh-pages` branch, which you can then set to make it a website.

**Reply by chuymtz** [reply]
Oh wow. Thanks for the quick response I really appreciate it. I will try it out. Again thanks

**Reply by rowanc1** [reply]
No problem, good luck writing! :)
